### PR TITLE
Add types for istanbuljs libraries

### PR DESCRIPTION
**Pull request by jason0x43**
This PR includes typings for the libraries in the [istanbuljs monorepo](https://github.com/istanbuljs/istanbuljs/tree/master/packages):

* istanbul-lib-coverage
* istanbul-lib-hook
* istanbul-lib-instrument
* istanbul-lib-report
* istanbul-lib-source-maps
* istanbul-reports

This PR includes typings for all the libraries mentioned above because their inter-dependence was going to make individual PRs a pain.

This is related to https://github.com/theintern/intern/issues/765

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/istanbul-reports/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        "istanbul-reports-tests.ts"
+    ],
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": false,
```
Comment (minestarks) [inline types/istanbul-reports/tsconfig.json]: Please, `"strictNullChecks": true` for all new packages

File: types/istanbul-lib-source-maps/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        "istanbul-lib-source-maps-tests.ts"
+    ],
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": false,
```
Comment (minestarks) [inline types/istanbul-lib-source-maps/tsconfig.json]: `"strictNullChecks": true`

File: types/istanbul-lib-report/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        "istanbul-lib-report-tests.ts"
+    ],
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": false,
```
Comment (minestarks) [inline types/istanbul-lib-report/tsconfig.json]: `"strictNullChecks": true`

File: types/istanbul-lib-instrument/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        "istanbul-lib-instrument-tests.ts"
+    ],
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": false,
```
Comment (minestarks) [inline types/istanbul-lib-instrument/tsconfig.json]: `"strictNullChecks": true`

File: types/istanbul-lib-hook/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        "istanbul-lib-hook-tests.ts"
+    ],
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": false,
```
Comment (minestarks) [inline types/istanbul-lib-hook/tsconfig.json]: `"strictNullChecks": true`

File: types/istanbul-lib-coverage/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        "istanbul-lib-coverage-tests.ts"
+    ],
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": false,
```
Comment (minestarks) [inline types/istanbul-lib-coverage/tsconfig.json]: `"strictNullChecks": true`

**Review by minestarks — CHANGES_REQUESTED** [review]
LGTM except for the `strictNullChecks` nitpick

**Review by minestarks — APPROVED** [review]

**Comment by jason0x43** [comment]
@RyanCavanaugh, @andy-ms -- just pinging you because it's been a couple days since this was merged but it hasn't shown up yet in the npm repo.

**Comment by andy-ms** [comment]
@jason0x43 Thanks, fixed.

**Pull request merged**